snapcraft/internal/yaml.py

```diff
@@ -132,10 +128,18 @@ def __init__(self, project_options=None):
 
         self.build_tools = self.data.get('build-packages', [])
         self.build_tools.extend(project_options.additional_build_packages)
-
-        self._remote_parts = parts.get_remote_parts()
         self._process_parts()
 
+    @property
+    def part_names(self):
+        return self._part_names
+
+    @property
+    def _remote_parts(self):
+        if getattr(self, '_remote_parts_attr', None) is None:
+            self._remote_parts_attr = parts.get_remote_parts()
+        return self._remote_parts_attr
```

Comment:
well, not really, I think. The double underscore does mangling, but there's always a way to access the attribute from outside. And from inside it works the same as one underscore. So I don't like it too much. I prefer just to rely on our own style in which one underscore means internal, and we try to respect it as much as possible.